arch/risc-v/src/mpfs/Make.defs

```diff
@@ -89,3 +89,39 @@ endif
 ifeq (${CONFIG_MPFS_BOOTLOADER},y)
 CHIP_CSRCS += mpfs_cache.c
 endif
+
+ifeq (${CONFIG_MPFS_OPENSBI},y)
+CHIP_ASRCS += mpfs_opensbi_utils.S
+CHIP_CSRCS += mpfs_opensbi.c
+
+OPENSBI_UNPACK  = opensbi-3rdparty
+OPENSBI_COMMIT  = 69d7e536138ae71a24028ca849d401a4d64d564b
+OPENSBI_URL     = https://github.com/riscv-software-src/opensbi/tarball
+OPENSBI_TARBALL = opensbi.tar.gz
+OPENSBI_DIR     = riscv-software-src-opensbi-69d7e53
+
+$(OPENSBI_TARBALL):
+	$(Q) echo "Downloading: OpenSBI"
+	$(Q) curl -L $(OPENSBI_URL)/$(OPENSBI_COMMIT) -o chip/$(OPENSBI_TARBALL)
+
+chip/$(OPENSBI_UNPACK): $(OPENSBI_TARBALL)
+	$(Q) echo "Unpacking: OpenSBI"
+	$(Q) tar xzf chip/$(OPENSBI_TARBALL) --directory chip/
+	$(Q) mv chip/$(OPENSBI_DIR)* chip/$(OPENSBI_UNPACK)
+	$(Q) touch chip/$(OPENSBI_UNPACK)
+	$(Q) (cd chip/$(OPENSBI_UNPACK); make;)
```

Comment:
I think in the old commit some static libraries were used, are we building them here?
We usually don't build packages during `context`.

Reply:
Ok, how should it be built? I think it should not be compiled within the nuttx itself, but as a static lib.

Reply:
Is there a static library to directly download?  If there is, we can just pick that and add it to `EXTRA_LIBS`.
If we need to build it ourselves we can also do that, we have examples for [OpenAMP](https://github.com/apache/incubator-nuttx/tree/master/openamp) and [LittleFS](https://github.com/apache/incubator-nuttx/tree/master/fs/littlefs).

Reply:
I created a new patch, no static libs to download. Now it's like openamp, but maybe it needs to be under riscv -architecture?

Reply:
Yes, I too think it should be under riscv folder.
Also, you can just add the files to `CSCR` and include the opensbi/Make.defs file into arch/risc-v/src/Makefile. This way you won't have to implement other common targets and add the opensbi directory to the build.  I think this way it will be cleaner.

Reply:
Looks like it gets really messy if put under arch/risc-v/src/opensbi, that dir is wiped out on make distclean, it has no 'context' ((CONTEXTDIRS = boards fs $(APPDIR) $(ARCH_SRC) as seen on tools/Directories.mk), adding it there probably causes more confusion..

Reply:
ARCH_SRC is actually the src directory under the selected arch.  So in this case it is arch/riscv/src.  The `context` target there is a double colon target, so we can add another double colon target under arch/riscv/src/opensbi and include the Make.defs in our arch/riscv/src/Makefile.

Reply:
Regarding `distclean`, what should be wiped out is only the downloaded part.

Reply:
Thanks, will update soon

Reply:
Still the licence issue present, not sure what way to go

Reply:
OpenSBI is distributed under a compatible license (BSD-2).  If we are using their files or code we should keep the original license and add an entry in our LICENSE file.

Reply:
Tried to update those properly on the latest commit. It's only the mpfs_opensbi_utils.S file where OpenSBI reference code is borrowed. Of course the library itself should have the proper license files built-in.